### PR TITLE
fix: native push permission status match typescript enum value

### PR DESCRIPTION
**Pull request by levibostian**
Closes: https://github.com/customerio/customerio-reactnative/issues/115

Alternative solution to [this pr](https://github.com/customerio/customerio-reactnative/pull/116). I decided to create a separate PR to demonstrate an idea that I had instead of modifying the branch in [this pr](https://github.com/customerio/customerio-reactnative/pull/116). 

It's recommended that only 1 PR is merged by the team.

### CODE REVIEW

**Review by mrehan27 — APPROVED** [review]
Looks good to me 👍🏻

**Comment by levibostian** [comment]
PR has been QA tested. 

* Installed RN SDK branch into Ami app. 
* Got push permission status - value is `NotDetermined` (not `Notdetermined`)
* Denied push permission. Got status - value is `Denied`
* Accepted push permission. Got status - value is `Granted`. 

All strings match the strings in [the type enum](https://github.com/customerio/customerio-reactnative/blob/main/src/types/PushPermissionStatus.tsx) that we expect.

File: ios/StringExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
+//  StringExtensions.swift
+//  CustomerioReactnative
+//
+//  Created by Levi Bostian on 3/22/23.
+//  Copyright © 2023 Facebook. All rights reserved.
```
[the remark this answers is not in the document]
Comment (levibostian) [reply ios/StringExtensions.swift:6]: Good catch. 

Sounds like this project would benefit from `swiftlint` like the iOS SDK does. Automatically take care of PR suggestions such as this one. 

[I created a ticket](https://github.com/customerio/issues/issues/9677) for that. In the mean time, I'll remove the header.

**Comment by mrehan27** [comment]
Created [PR](https://github.com/customerio/docs/pull/1066) for doc updates

**Pull request merged**